### PR TITLE
decorate a class

### PR DESCRIPTION
**Pull request by rbreva**
### Decorate a class

- Create a class Nameable
- Prepare base Decorator
- Prepare CapitalizeDecorator and TrimmerDecorator

thanks for the review

Mauricio Mendoza

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,8 +1,11 @@
+require_relative 'nameable'
+
 class Person
```
Comment (mugishasam123) [inline person.rb]: - [ ] Please try to make the person class inherit from nameable and create the `correct_name` method in the person class.

File: person.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
   attr_accessor :name, :age
   attr_reader :id
 
   def initialize(age, name: 'Unkown', parent_permission: true)
+    super
     @age = age
     @name = name
     @parent_permission = parent_permission
```
Comment (mugishasam123) [inline person.rb]: - [ ] Please try to leverage your person class to take the required arguments as the project guidelines state to make your class pass tests.

errors while testing :
image

**Review by mugishasam123 — CHANGES_REQUESTED** [review]
Hi @rbreva  👋 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
alt
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Descriptive PR summary 🚀 
- linters are passing 👍 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MaganaAsiati — APPROVED** [review]
# Status: Approved [✔️](https://emojipedia.org/check-mark/)
## Hi @rbreva  , I'm @MaganaAsiati , your reviewer for this PR  :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


clap

## Highlights
- descriptive README file  :heavy_check_mark:👨🏻‍💻
- Your commits should start with a capital letter :heavy_check_mark:👨🏻‍💻


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future.
- [ ] [OPTIONAL] kindly  Add a meaningful description in your pull request in the future which give more information about what you implement in this milestone :heavy_check_mark:👨🏻‍💻


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Cheers and Happy coding!👏👏👏

**Pull request merged**